### DISCUSSION TITLE
I found an invalid date string that moment validates as correct

### DISCUSSION THREAD

**Discussion by gyrevik** [moment/moment#6418] [Q&A] [answered]
**Describe the bug**
I found an invalid date string that moment validates as correct
**To Reproduce**
1. Run the following code:
alert(moment("/6/2012", 'MM/DD/YYYY').isValid()); // returns true

**Expected behavior**
Return false.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser:  Edge
 - Version 109.0.1518.55 (Official build) (64-bit) -- Edge up to date as of 1/17/2023

**Moment-specific environment**
React running on node v14.19.0

* The time zone setting of the machine the code is running on
Eastern US

* The time and date at which the code was run
After 11 PM 1/17/2023

* Other libraries in use (TypeScript, Immutable.js, etc)
A lot

Please run the following code in your environment and include the output:
```
console.log((new Date()).toString())
console.log((new Date()).toLocaleString())
console.log((new Date()).getTimezoneOffset())
console.log(navigator.userAgent)
console.log(moment.version)
```

Tue Jan 17 2023 23:35:50 GMT-0500 (Eastern Standard Time)
tabs/src/dashboard/filterPanel/components/datePicker.tsx:84
1/17/2023, 11:35:50 PM
tabs/src/dashboard/filterPanel/components/datePicker.tsx:85
300
tabs/src/dashboard/filterPanel/components/datePicker.tsx:86
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36 Edg/109.0.1518.55
tabs/src/dashboard/filterPanel/components/datePicker.tsx:87
2.29.4

**Additional context**
First line of the below fiddle produces the problem:
http://jsfiddle.net/scyk1nLm/

**Comment by albajarazeth** [comment]
Hi! I would like to work on this issue, can I pick it up?

**Comment by gyrevik** [comment]
I don't know if that's up to me but it's all good if you ask me.

**Comment by mattjohnsonpint** [comment, marked as answer]
Thanks for the reproduction! The two-argument constructor uses lenient parsing. For validation, I'd enable strict mode: `moment('/6/2012', 'MM/DD/YYYY', true).isValid()`, which returns `false`. The small third argument makes the important difference here.
